tripleblindmarket/private-kit
# E2E Transfer Flow

#### Description:
Wires up the flow for the e2e. This sets them against our apiary server to handle as the api contracts are expected. 

This is all behind a feature flag. When the staging server is set up, we can add it to the HA list in a way that we can dynamically route the upload to HA's based on their api url's in the yaml. For now, this is hard coded to the apiary. 

1. Select an HA
2. Enter code
3. Consent to both agreements
4. Upload data

https://www.figma.com/file/79OwaKqHvSeriExpYzAvVN/PATHS-%2F-Exploration-%2F-April-%26-May-2020?node-id=3600%3A26288

#### Screenshots:

<img width="571" alt="image" src="https://user-images.githubusercontent.com/25315679/83298648-d3f50900-a1c2-11ea-994f-a6f6b5072b83.png">
<img width="571" alt="image" src="https://user-images.githubusercontent.com/25315679/83298656-d6576300-a1c2-11ea-9050-aa31a9872299.png">
<img width="571" alt="image" src="https://user-images.githubusercontent.com/25315679/83298677-dc4d4400-a1c2-11ea-9c9f-a133d8fa94c3.png">
<img width="571" alt="image" src="https://user-images.githubusercontent.com/25315679/83298686-df483480-a1c2-11ea-9877-59834fa25792.png">
<img width="571" alt="image" src="https://user-images.githubusercontent.com/25315679/83298693-e1aa8e80-a1c2-11ea-914f-6e80d8f6b9fd.png">
<img width="571" alt="image" src="https://user-images.githubusercontent.com/25315679/83298699-e40ce880-a1c2-11ea-83c8-92a269929116.png">
<img width="571" alt="image" src="https://user-images.githubusercontent.com/25315679/83298702-e66f4280-a1c2-11ea-97b9-6878af214684.png">

![2020-05-29 15 41 51](https://user-images.githubusercontent.com/25315679/83298906-40700800-a1c3-11ea-8683-75aec9a87eec.gif)

<img width="559" alt="image" src="https://user-images.githubusercontent.com/25315679/83316658-18010180-a1f5-11ea-9cd5-ce22031b2636.png">

>What's with all these funky `flexGrows` ? 
Accessibility hacks for scroll views for content: 
<img width="877" alt="image" src="https://user-images.githubusercontent.com/25315679/83315977-60b6bb80-a1f1-11ea-9ac9-fc348b3210bb.png">
 

#### How to test:
The primary thing to test with these is the UI/UX of the flows. We're awaiting the staging server to be able to handle a full e2e test with this.

- Test the primary flow, of going through the flow screen by screen. With the mock api's, any code will work right now.

- To test the fail cases, disable wifi while going through the screens

- To test the failed code (because of a _bad_ code, rather than a generic failure), you've gotta either change the endpoint, or change the condition in the code screen to fail on the 200. This will be easier when we have a working endpoint.

- This should be good across a variety of phone sizes. I've tested this on iPhone 11, 8 & 6, and Pixel 2
